querier/src/table/mod.rs

```diff
@@ -895,20 +763,55 @@ mod tests {
         )
     }
 
-    /// Invokes querier_table.chunks modeling the ingester sending `next_response`
-    async fn chunks_with_ingester_partition(
-        querier_table: &QuerierTable,
-        ingester_partition: &IngesterPartition,
-    ) -> Vec<Arc<dyn QueryChunk>> {
-        let pred = Predicate::default();
-        querier_table
-            .ingester_connection
-            .as_any()
-            .downcast_ref::<MockIngesterConnection>()
-            .unwrap()
-            .next_response(Ok(vec![ingester_partition.clone()]));
+    /// A `QuerierTable` and some number of `IngesterPartitions` that
+    /// are fed to the ingester connection on the next call to
+    /// `chunks()`
+    struct TestQuerierTable {
+        // The underling table
+        querier_table: QuerierTable,
+        /// Ingester partitions
+        ingester_partitions: Vec<IngesterPartition>,
+    }
+
+    impl TestQuerierTable {
+        /// Create a new wrapped [`QuerierTable`]
+        async fn new(catalog: &Arc<TestCatalog>, table: &Arc<TestTable>) -> Self {
+            Self {
+                querier_table: querier_table(catalog, table).await,
+                ingester_partitions: vec![],
+            }
+        }
+
+        /// Return a reference to the inner table
+        fn inner(&self) -> &QuerierTable {
+            &self.querier_table
+        }
+
+        /// add the `ingester_partition` to the ingester response processed by the table
+        fn with_partition(mut self, ingester_partition: IngesterPartition) -> Self {
```

Comment:
```suggestion
        fn with_ingester_partition(mut self, ingester_partition: IngesterPartition) -> Self {
```

Just to make it really clear which state you're changing/mocking. Otherwise `table.with_partition` might be misread a bit.

Reply:
in 5a1b34a14